app-libs/stf/src/stf_sgx.rs

```diff
@@ -135,11 +135,12 @@ where
 
 	fn execute_call(
 		state: &mut State,
+		shard: &ShardIdentifier,
 		call: Call,
 		calls: &mut Vec<OpaqueCall>,
 		node_metadata_repo: Arc<NodeMetadataRepository>,
 	) -> Result<(), Self::Error> {
-		state.execute_with(|| call.execute(calls, node_metadata_repo))
+		state.execute_with(|| call.execute(shard, calls, node_metadata_repo))
```

Comment:
Why don't pass `shard` similar like `TrustedCall::balance_unshield(account_incognito, beneficiary, value, shard)` ? Instead of directly pass in fn() as one extra param?

Reply:
> Why don't pass `shard` similar like `TrustedCall::balance_unshield(account_incognito, beneficiary, value, shard)` , instead of directly pass in fn() as one extra param?

because state is a per shard thing. In fact `execute_trusted_call_on_stf` has `shard` as its parameter, but then gets dropped after `verify_signature`, and IMO `execute_calls` should retain this information, so the shard in `balance_unshield` is redundant if the shard is meant to be the state-bound shard => we'll propose an upstream PR later